EVES/drafts/EVES-005/eves-005.md

```diff
@@ -76,10 +76,26 @@ The consumer uses metadata search or similar services to identify an asset of in
 
 At this point, the contract is fully settled and the consumer can use the asset.
 
-### 5. Fee Payment
+### 5. Fee and Payment Report
 
-On a regular basis, the providers send the accumulated fees to the operator.
-To ensure the fees are correctly calculated without publishing business statistics, a provider should construct a zero knowledge proof for the accumulated amount being correct based on the submitted hashes.
+#### 5.1. Asset Fee
+
+1. After a defined period, the provider compiles all completed contracts and generates a cumulative bill.
+2. The provider sends this bill to the respective customers.
+3. The customers review the bill and make payments.
+
+#### 5.2. Operator Fee
+
+In addition to handling customer payments, providers are responsible for paying the accumulated fees to the operator.
+
+To ensure accurate fee calculations while maintaining business confidentiality:
+
+1. Providers must construct a Zero-Knowledge Proof (ZKP) to verify the correctness of the accumulated fees based on submitted hashes.
+   These hashes should reference the cumulative bills issued to customers.
```

Comment:
And this is where it gets complicated and bricks the existing spec: we do not sign bills digitally and do not publish hashes. That would possibly create privacy issues. We need to figure out how "pay-per-use" service contracts should work, since the contract does not contain a specific price that we could base market fees on.

Do we need/want to introduce invoice VCs?